packages/xrpl/src/models/transactions/common.ts

```diff
@@ -329,6 +341,35 @@ export function isArray<T = unknown>(input: unknown): input is T[] {
   return input != null && Array.isArray(input)
 }
 
+/**
+ * Verify the input is a valid ledger entry id.
+ *
+ * @param input - The object to validate.
+ * @returns Whether the input is a valid ledger entry id.
+ */
+export function isLedgerEntryId(input: unknown): input is string {
+  return isString(input) && isHex(input) && input.length === SHA_512_HALF_LENGTH
+}
+
+/**
+ * Validate input is non-empty hex string of up to a certain length.
+ *
+ * @param input - The metadata to validate.
+ * @param lengthUpto - The maximum length of the hex string.
+ * @returns Whether the input is a valid non-empty hex string up to the specified length.
+ */
+export function validateHexMetadata(
```

Comment:
Maybe we should add an issue/to-do to add these checks to transactions outside of lending protocol as well